### PR TITLE
Polaris super consumer

### PR DESCRIPTION
**Pull request by elaubsch**
This PR introduces a Polaris "super consumer" that handles spot detection and cell segmentation for singleplex FISH analysis. 

- The consumer that handles spot detection tasks - previously called the `PolarisConsumer` - has been refactored to the `SpotDetectionConsumer`. 
- The label detection model has been removed from the segmentation consumer and the channel handling for nuclear and whole-cell segmentation is now done manually.
- The `MAX_SCALE` variable has been increased from 3 to 10 to allow for segmentation of 100x images, which are typical for FISH experiments.

### CODE REVIEW

File: redis_consumer/consumers/segmentation_consumer.py [hunk 1]
```diff
@@ -104,13 +63,15 @@ def _consume(self, redis_hash):
...
+        if len(np.shape(image)) == 3:
+            image = np.expand_dims(image, axis=-1)  # add a channel dimension
 
-        # Validate input image
-        if hvals.get('channels'):
-            channels = [int(c) for c in hvals.get('channels').split(',')]
-        else:
-            channels = None
+        rank = 4  # (b,x,y,c)
+        channel_axis = image.shape[1:].index(min(image.shape[1:])) + 1
```
[the remark this answers is not in the document]
Comment (elaubsch) [reply redis_consumer/consumers/segmentation_consumer.py:114]: We perform the same manipulation in the `validate_model_input` method of the base consumer class [here](https://github.com/vanvalenlab/kiosk-redis-consumer/blob/94cdf9be1cfa561c2b77454b74c342a39787f1ab/redis_consumer/consumers/base_consumer.py#L302), happy to re-evaluate both cases but it seems work stably for the base consumer

[the remark this answers is not in the document]
Comment (msschwartz21) [reply redis_consumer/consumers/segmentation_consumer.py:114]: Ok I can live with that

File: redis_consumer/consumers/polaris_consumer.py [hunk 2]
```diff
@@ -23,64 +23,295 @@
...
+        image_hvals = {
+            'identity_upload': self.name,
+            'input_file_name': upload_file_name,
+            'original_name': image_fname,
+            'status': 'new',
+            'created_at': current_timestamp,
+            'updated_at': current_timestamp,
+            'url': upload_file_url,
+            'channels': channels,
+            'scale': 0.38}  # TODO don't hardcode
```
Comment (msschwartz21) [inline redis_consumer/consumers/polaris_consumer.py:81]: Should this TODO be addressed?

File: redis_consumer/consumers/polaris_consumer.py [hunk 3]
```diff
@@ -23,64 +23,295 @@
...
+
+            remaining_hashes -= finished_hashes
+            time.sleep(settings.INTERVAL)
+
+        if segmentation_type == 'cell culture':
+            for key in segmentation_dict.keys():
+                labeled_im = np.array(segmentation_dict[key])
+                labeled_im = np.squeeze(labeled_im, -1)
+                labeled_im = np.moveaxis(labeled_im, 0, 2)
+                segmentation_results.append(labeled_im)
```
Comment (msschwartz21) [inline redis_consumer/consumers/polaris_consumer.py:222]: What is happening here that can't get handled above inside of the while loop?

File: redis_consumer/consumers/polaris_consumer.py [hunk 4]
```diff
@@ -23,64 +23,295 @@
...
+        image_hvals = {
+            'identity_upload': self.name,
+            'input_file_name': upload_file_name,
+            'original_name': image_fname,
+            'status': 'new',
+            'created_at': current_timestamp,
+            'updated_at': current_timestamp,
+            'url': upload_file_url,
+            'channels': channels,
+            'scale': 0.38}  # scaling not supported for spots model
```
Comment (msschwartz21) [inline redis_consumer/consumers/polaris_consumer.py:81]: Can we move this into settings with the other polaris parameters?

File: redis_consumer/consumers/polaris_consumer.py [hunk 5]
```diff
@@ -102,86 +333,35 @@ def _consume(self, redis_hash):
...
-        dest, output_url = self.save_output(results, image, save_name)
+            dest, output_url = self.save_output(res, hvals)
 
         # Update redis with the final results
         end = timeit.default_timer()
         self.update_key(redis_hash, {
             'status': self.final_status,
             'output_url': output_url,
-            'upload_time': end - _,
+            'upload_time': end,  # - _
```
Comment (msschwartz21) [inline redis_consumer/consumers/polaris_consumer.py:364]: What's up with this?

File: redis_consumer/consumers/polaris_consumer_test.py [hunk 6]
```diff
@@ -66,31 +158,35 @@ def test__consume_finished_status(self, redis_client):
...
-        )
-
-        mocker.patch.object(consumer, 'get_grpc_app', lambda *x, **_: mock_app)
-        mocker.patch.object(consumer, 'get_image_scale', lambda *x, **_: 1)
-        mocker.patch.object(consumer, 'validate_model_input', lambda *x, **_: x[0])
-        mocker.patch.object(consumer, 'detect_dimension_order', lambda *x, **_: 'YXC')
-
+        # consume with segmentation and spot detection
+        empty_data = {'input_file_name': 'file.tiff',
+                      'segmentation_type': 'cell culture'}
```
Comment (msschwartz21) [inline redis_consumer/consumers/polaris_consumer_test.py:168]: Should there also be a test case for mesmer segmentation or are the workflows similar enough for it to be unnecessary?

**Review by msschwartz21 — COMMENTED** [review]
This looks great! Thanks for all of the hard work!

Let me know what you think about the couple of last comments I left and then I'll approve.

File: redis_consumer/consumers/polaris_consumer.py (hunk 5 above)
Comment (elaubsch) [reply redis_consumer/consumers/polaris_consumer.py:364]: Oops I think I was debugging something, I'll revert the change

File: redis_consumer/consumers/polaris_consumer_test.py (hunk 6 above)
Comment (elaubsch) [reply redis_consumer/consumers/polaris_consumer_test.py:168]: They are pretty similar but it's an easy add

**Review by msschwartz21 — APPROVED** [review]

**Pull request merged**